File: 백준/Silver/1302. 베스트셀러/베스트셀러.py
import sys

data = sys.stdin.read().splitlines()

n = int(data[0])
booklist = data[1:]
check = {}

for name in booklist:
  if name in check.keys():
    check[name] += 1
  else:
    check[name] = 1
 
max_val = max(check.values())
max_keys = [k for k, v in check.items() if v == max_val]

print(min(max_keys))

File: 백준/Silver/1302. 베스트셀러/베스트셀러.cc
#include <iostream>
#include <map>
#include <string>
using namespace std;

int main() {
  int n;
  cin >> n;
  map<string, int> check;

  for (int i=0; i<n; i++) {
    string name;
    cin >> name;
    check[name]++;
  }

  int max_val = 0;
  string max_key;

  for (auto &p : check) {
    if (p.second > max_val) {
      max_val = p.second;
      max_key = p.first;
    }
  }

  cout << max_key << "\n";
  return 0;
}